quora-service/src/main/java/com/upgrad/quora/service/business/AuthenticationService.java

```diff
@@ -60,4 +62,23 @@ public UserAuthTokenEntity authenticate(final String username, final String pass
             throw new AuthenticationFailedException("ATH-002", "Password failed");
         }
     }
+
+    /**
+     * Method to delete the authorization token
+     * @param authorizationToken
+     * @return
+     * @throws SignOutRestrictedException
+     */
+    @Transactional(propagation = Propagation.REQUIRED)
+    public String deleteToken(final String authorizationToken) throws SignOutRestrictedException {
+        //Check if the user is not logged in
+        UserAuthTokenEntity userAuthTokenEntity = userDao.getUserAuthToken(authorizationToken);
+        if (userAuthTokenEntity == null) {
+            throw new SignOutRestrictedException("SGR-001", "User is not Signed in");
+        }
+        String uuid = userAuthTokenEntity.getUuid();
+        userDao.deleteAuthToken(userAuthTokenEntity);
```

Comment:
The method says that if the token is valid we need to update logoutAt time   which i didn't see in the code above can u plz explain? and also why deleteAuthtoken ?

Reply:
Changed.

Reply:
If the access token provided by the user is valid, update the LogoutAt time of the user in the database and return the 'uuid' of the signed out user from 'users' table and message 'SIGNED OUT SUCCESSFULLY' in the JSON response with the corresponding HTTP status.

Please address this requirement